ci_framework/plugins/action/discover_latest_image.py

```diff
@@ -0,0 +1,113 @@
+# Copyright Red Hat, Inc.
+# Apache License Version 2.0 (see LICENSE)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+action: discover_latest_image
+
+short_description: Discover the latest image available on remote server.
+
+description:
+- Discover latest image matching a specific pattern on remote server.
+
+options: Please refer to ansible.builtin.uri options
+
+'''
+
+EXAMPLES = r'''
+- name: Get latest CentOS 9 Stream image
+  register: discovered_images
+  discover_latest_image:
+    base_url: "https://cloud.centos.org/centos/{{ ansible_distribution_major_version }}-stream/x86_64/images/"
+    image_prefix: "CentOS-Stream-GenericCloud-"
+'''
+
+RETURN = r'''
+success:
+    description: Status of the module
+    type: bool
+    returned: always
+    sample: True
+data:
+    description: Dict grouping the various data the action fetches
+    type: dict
+    returned: always
+'''
+
+from ansible.errors import AnsibleError
+from ansible.plugins.action import ActionBase
+from ansible.module_utils.basic import missing_required_lib
+
+import re
+
+try:
+    from bs4 import BeautifulSoup
+except ImportError as imp_exc:
+    BS4_IMPORT = imp_exc
+else:
+    BS4_IMPORT = None
+
+
+class ActionModule(ActionBase):
+    def run(self, tmp=None, task_vars=None):
+        super(ActionModule, self).run(tmp, task_vars)
+        module_args = self._task.args.copy()
+
+        if BS4_IMPORT:
+            raise AnsibleError(missing_required_lib('beautifulsoup4'))
+
+        if 'image_prefix' not in module_args:
+            raise AnsibleError('"image_prefix" parameter is mandatory')
+
+        img_prefix = module_args.pop('image_prefix')
+
+        result = {
```

Comment:
Why not initializing these when used? At a first glance I expected this to be used in the following lines but it's used at near the end, when the whole result can be created in line 105